src/pages/main/Main.tsx

```diff
@@ -1,5 +1,117 @@
+import { Badge, Box, Button, Stack, Typography } from '@mui/material';
+import { useSearchParams } from 'react-router-dom';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import TuneIcon from '@mui/icons-material/Tune';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import MainModal from './components/Modal';
+import { mainStyles } from './index.styles';
+import DataFetchError from '../../components/shared/DataFetchError';
+import LoadingPrimary from '../../components/loader/LoadingPrimary';
+import AccommodationCard from '../../components/card/AccommodationCard';
+import { Accommodation, DefaultSearchParamsType } from '../../types/accommodation.types';
+import useGetAccommodationsQuery from '../../api/queries/main/useGetAccommodationsQuery';
+
+const defaultSearchParams: DefaultSearchParamsType = {
+  minPrice: '0',
+  maxPrice: '0',
+  minRooms: '0',
+  minPeople: '0',
+  orderByPrice: 'any',
+  orderByPeople: 'any',
+  orderByRoom: 'any',
+};
+
 function Main() {
-  return <div>Main</div>;
+  const [open, setOpen] = useState(false);
+  const [invisible, setInvisible] = useState(false);
+  const [priceRange, setPriceRange] = useState({
+    totalMaxPrice: 0,
+    totalMinPrice: 0,
+  });
+
+  const [searchParams, setSearchParams] = useSearchParams(defaultSearchParams);
+
+  const searchParamsAsObject = useMemo(() => {
+    const values: Record<string, string> = {};
+    return values;
+  }, []);
+
+  for (const [key, value] of searchParams.entries()) {
```

Comment:
you can move this logic inside `useMemo` a little above to avoid unnecessary computations